source-http-ingest/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+        Response::Opened {
+            explicit_acknowledgements: true,
+        },
+        &mut stdout,
+    )
+    .await?;
+
+    server::run_server(config, typed_bindings, stdin, stdout).await
+}
+
+async fn do_apply(mut stdout: io::Stdout) -> anyhow::Result<()> {
+    // There's nothing to apply
+    write_capture_response(
+        Response::Applied {
+            action_description: String::new(),
+        },
+        &mut stdout,
+    )
+    .await
+}
+
+async fn do_spec(mut stdout: io::Stdout) -> anyhow::Result<()> {
+    let config_schema = to_raw_value(&schema_for::<EndpointConfig>())?;
+    let resource_config_schema = to_raw_value(&schema_for::<ResourceConfig>())?;
+    let response = Response::Spec {
+        documentation_url: "https://go.estuary.dev/http-ingest".to_string(),
+        config_schema,
+        resource_config_schema,
+        oauth2: None,
+    };
+    write_capture_response(response, &mut stdout).await
+}
+
+async fn do_discover(config: RawValue, mut stdout: io::Stdout) -> anyhow::Result<()> {
+    // make sure we can parse the config, just as an extra sanity check
+    let _ =
+        serde_json::from_str::<EndpointConfig>(config.get()).context("parsing endpoint config")?;
+    let bindings = vec![discovered_webhook_collection()];
+    let response = Response::Discovered { bindings };
+    write_capture_response(response, &mut stdout).await
+}
+
+async fn do_validate(
+    config: RawValue,
+    bindings: Vec<ValidateBinding>,
+    mut stdout: io::Stdout,
+) -> anyhow::Result<()> {
+    let config = serde_json::from_str::<EndpointConfig>(config.get())
+        .context("deserializing endpoint config")?;
+    let mut output = Vec::with_capacity(bindings.len());
+    let mut typed_bindings = Vec::with_capacity(bindings.len());
+    for ValidateBinding {
+        collection,
+        resource_config,
+    } in bindings
+    {
+        let resource_config = serde_json::from_str::<ResourceConfig>(resource_config.get())
+            .context("deserializing resource config")?;
+
+        let resource_path = if let Some(path) = resource_config.path.as_ref() {
+            vec![server::ensure_slash_prefix(path.as_str().trim())]
+        } else {
+            vec![collection.name.clone()]
+        };
+
+        output.push(ValidatedBinding {
+            resource_path: resource_path.clone(),
+        });
+        typed_bindings.push(Binding {
+            collection,
+            resource_path,
+            resource_config,
+        });
+    }
+
+    // Check to make sure we can successfully create an openapi spec
+    server::openapi_spec(&config, &typed_bindings)
+        .context("cannot create openapi spec from bindings")?;
+    let response = Response::Validated { bindings: output };
+    write_capture_response(response, &mut stdout).await
+}
+
+pub async fn read_capture_request(stdin: &mut io::BufReader<io::Stdin>) -> anyhow::Result<Request> {
+    let mut buf = String::with_capacity(4096);
```

Comment:
4096 seems plenty long, but I know _materialization_ bindings are pretty detailed. I'm not sure what capture bindings look like exactly off hand, so this probably isn't an issue. But if they have a ton of bindings I'm not sure what would happen here.

Reply:
The `4096` is actually just the _initial_ capacity. It'll grow automatically if it needs to. I just tend to be in the habit of giving explicit initial sizes to buffers. But in _this_ case, it's probably not providing any benefit since we're likely to read an entire request's worth of bytes in a single go, anyway.